src/sentry/utils/avatar.py

```diff
@@ -27,3 +28,51 @@ def get_gravatar_url(email, size=None, default='mm'):
         gravatar_url += "?" + urllib.urlencode(properties)
 
     return gravatar_url
+
+
+LETAR_COLORS = {
+    'blue': '#25A6F7',
+    'green': '#6FBA57',
+    'orange': '#EC5B33',
+    'red': '#E35141',
+    'pink': '#E56AA6',
+    'purple': '#836CC2'
+}
+
+
+def hash_user_identifier(identifier):
+    return sum(map(ord, identifier))
+
+
+def get_letar_color(user):
+    # If you change anything with color selection,
+    # make sure you also change it in letar.jsx
+    hashed_id = hash_user_identifier(user.get_label())
+    color = 'purple'
+    if hashed_id % 6 == 0:
+        color = 'blue'
+    elif hashed_id % 5 == 0:
+        color = 'green'
+    elif hashed_id % 4 == 0:
+        color = 'orange'
+    elif hashed_id % 3 == 0:
+        color = 'red'
+    elif hashed_id % 2 == 0:
+        color = 'pink'
+    return LETAR_COLORS[color]
+
+
+def get_letar(user, size=None):
+    display_name = user.get_display_name() or '?'
+    names = display_name.split(' ')
+    initials = '%s%s' % (names[0][0], names[-1][0] if len(names) > 1 else '')
+    initials = escape(initials.upper())
+    color = get_letar_color(user)
+    size_attrs = 'height="%s" width="%s"' % (size, size) if size else ''
+    return (
+        u'<svg viewBox="0 0 120 120" xmlns="http://www.w3.org/2000/svg" {size_attrs}>'
+        '<rect x="0" y="0" width="120" height="120" rx="15" ry="15" fill={color}></rect>'
+        '<text x="50%" y="50%" font-size="65" dominant-baseline="central" text-anchor="middle" fill="#FFFFFF">'
```

Comment:
this is the correct way to do this, but the version of react that we're using doesn't support the dominant-baseline svg property (it should soon, see https://github.com/facebook/react/pull/5714). adding it as a style property has the same effect, so just doing that for now. i should maybe add a TODO to update the jsx version whenever we upgrade react though